cargo-pgrx/src/command/schema.rs

```diff
@@ -376,195 +295,272 @@ pub(crate) fn generate_schema(
+        command.arg(user_package);
+    }
 
-    let mut symbols_to_stub = HashSet::new();
-    for export in postmaster_exports {
-        let name = std::str::from_utf8(export.name())?.to_string();
-        #[cfg(target_os = "macos")]
-        let name = {
-            // Mac will prefix symbols with `_` automatically, so we remove it to avoid getting
-            // two.
-            let mut name = name;
-            let rename = name.split_off(1);
-            assert_eq!(name, "_");
-            rename
-        };
-        symbols_to_stub.insert(name);
+    if let Some(user_manifest_path) = user_manifest_path.as_ref() {
+        command.arg("--manifest-path");
+        command.arg(user_manifest_path.as_ref());
     }
 
-    tracing::debug!("Creating stub of appropriate PostgreSQL symbols");
-    PgrxPgSysStub::from_symbols(&symbols_to_stub)?.write_to_file(&postmaster_stub_file)?;
+    command.args(profile.cargo_args());
 
-    let mut so_rustc_invocation = crate::env::rustc();
-    so_rustc_invocation.stderr(Stdio::inherit());
+    if let Some(log_level) = &log_level {
+        command.env("RUST_LOG", log_level);
+    }
 
-    if let Ok(rustc_flags_str) = std::env::var("RUSTFLAGS") {
-        let rustc_flags = rustc_flags_str.split_whitespace().collect::<Vec<_>>();
-        if !rustc_flags.is_empty() {
-            so_rustc_invocation.args(rustc_flags);
-        }
+    if !features_arg.trim().is_empty() {
+        command.arg("--features");
+        command.arg(&features_arg);
+    }
+
+    if features.no_default_features {
+        command.arg("--no-default-features");
+    }
+
+    if features.all_features {
+        command.arg("--all-features");
     }
 
-    so_rustc_invocation.args([
-        "--crate-type",
-        "cdylib",
-        "-o",
-        postmaster_stub_built
-            .to_str()
-            .ok_or(eyre!("could not call postmaster_stub_built.to_str()"))?,
-        postmaster_stub_file
-            .to_str()
-            .ok_or(eyre!("could not call postmaster_stub_file.to_str()"))?,
-    ]);
-
-    let so_rustc_invocation_str = format!("{:?}", so_rustc_invocation);
-    tracing::debug!(command = %so_rustc_invocation_str, "Running");
-    let output = so_rustc_invocation.output().wrap_err_with(|| {
-        eyre!("could not invoke `rustc` on {}", &postmaster_stub_file.display())
-    })?;
-
-    let code = output.status.code().ok_or(eyre!("could not get status code of build"))?;
-    tracing::trace!(status_code = %code, command = %so_rustc_invocation_str, "Finished");
-    if code != 0 {
-        return Err(eyre!("rustc exited with code {}", code));
+    for arg in flags.split_ascii_whitespace() {
+        command.arg(arg);
     }
 
-    std::fs::write(&postmaster_hash_file, postmaster_bin_hash)
-        .wrap_err("could not write postmaster stub hash")?;
+    command.arg("--");
 
-    Ok(postmaster_stub_built)
+    command.args(["--cfg", "pgrx_embed"]);
+
+    command.env("PGRX_EMBED", embed_path.as_ref());
```

Comment:
After examining the problem more closely I think we are better off with something that uses `RUSTFLAGS="--cfg pgrx_embed" cargo run --bin some_binary_name` as the primary command we use to execute the schema generation.

Reply:
`RUSTFLAGS="--cfg pgrx_embed"` leads to recompilation of all deps so I prefer `cargo rustc`, which only sets `cfg` for the bin target.

Reply:
Ah, true. That makes sense then. Can you add a comment about the reasons we are choosing `cargo rustc` over its alternatives, then, like that?